introduce START TRANSCATION FOR command

jasonmchan opened the pull request:
This commit changes the parser to support a new way to start a
transaction---the START TRANSACTION FOR command. The syntax is

```
START TRANSACTION [transaction_type] FOR ([transaction_args])
```

This command is identical to START TRANSACTION (or BEGIN), except it
first registers a parameterized transaction with the underlying
storage engines. The storage engines can use this information to make
scheduling decisions.

The purpose of the transaction type and the transaction args is to
enable us to map a transaction to a scheduling cluster. The type is an
unsigned integer, and the arguments are a list of literals.

Commit:
introduce START TRANSCATION FOR command

This commit changes the parser to support a new way to start a
transaction---the START TRANSACTION FOR command. The syntax is

```
START TRANSACTION [transaction_type] FOR ([transaction_args])
```

This command is identical to START TRANSACTION (or BEGIN), except it
first registers a parameterized transaction with the underlying
storage engines. The storage engines can use this information to make
scheduling decisions.

The purpose of the transaction type and the transaction args is to
enable us to map a transaction to a scheduling cluster. The type is an
unsigned integer, and the arguments are a list of literals.

audreyccheng reviewed (commented):
Changes so far lgtm. Is there a way to test the keyword and arguments will be parsed as expected?

jasonmchan commented:
> Changes so far lgtm. Is there a way to test the keyword and arguments will be parsed as expected?

I've updated the syntax to START TRANSACTION ... FOR ... to reflect that we're using this statement to pass transaction parameters and not necessarily do scheduling.

I also added a dummy implementation for the innobase handler, so it prints out the type and arguments when you issue a START TRANSACTION FOR statement.

audreyccheng reviewed (approved):
I guess we can merge these for now and you can make the Item change in a later PR?
